aws-api/src/main/java/com/amplifyframework/api/aws/AppSyncGraphQLRequest.java

```diff
@@ -0,0 +1,295 @@
+        return this.variables;
+    }
+
+    /**
+     * Used for setting a variable on the request.
+     * @param key variable name
+     * @param type type of variable value
+     * @param value variable value
+     */
+    public void setVariable(String key, String type, String value) {
+        this.variables.put(key, value);
+        this.variableTypes.put(key, type);
+    }
+
+    /**
+     * Used to add owner argument from authProvider if needed.
+     * @param authProvider CognitoUserPoolsAuthProvider for obtaining the username to set as the owner field.
+     * @throws ApiException if request requires owner argument and authProvider or authProvider.getUsername() is null.
+     */
+    public void setAuthProvider(CognitoUserPoolsAuthProvider authProvider) throws ApiException {
+        for (AuthRule authRule : modelSchema.getAuthRules()) {
+            if (isOwnerArgumentRequired(authRule.getOperationsOrDefault())) {
+                setVariable(authRule.getOwnerFieldOrDefault(), "String!", getUsername(authProvider));
+            }
+        }
+    }
+
+    private String getUsername(CognitoUserPoolsAuthProvider authProvider) throws ApiException {
+        if (authProvider == null) {
+            throw new ApiException(
+                    "Attempted to subscribe to a model with owner based authorization without a Cognito provider",
+                    "Did you add the AWSCognitoAuthPlugin to Amplify before configuring it?"
+            );
+        }
+        String username = authProvider.getUsername();
+        if (username == null) {
+            throw new ApiException(
+                    "Attempted to subscribe to a model with owner based authorization without a username",
+                    "Make sure that a user is logged in before subscribing to a model with owner based auth"
+            );
+        }
+        return username;
+    }
+
+    private boolean isOwnerArgumentRequired(List<ModelOperation> operations) {
+        if (SubscriptionType.ON_CREATE.equals(operationType) && operations.contains(ModelOperation.CREATE)) {
+            return true;
+        }
+        if (SubscriptionType.ON_UPDATE.equals(operationType) && operations.contains(ModelOperation.UPDATE)) {
+            return true;
+        }
+        if (SubscriptionType.ON_DELETE.equals(operationType) && operations.contains(ModelOperation.DELETE)) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     *  Returns String value used for GraphQL "query" in HTTP request body.
+     *
+     *  Sample return value:
+     *      subscription OnCreatePerson(owner: String!, nextToken: String) {
+     *            onCreatePerson(owner: $owner, nextToken: $nextToken) {
+     *               age dob first_name id last_name relationship owner
+     *            }
+     *       }
+     *
+     * @return String value used for GraphQL "query" in HTTP request body
+     */
+    @Override
+    public String getQuery() {
```

Comment:
I noticed that `nextToken` was removed as an input from the query in one of the test cases (`query-person-by-predicate.txt `) and I don't see that accounted for anywhere. Is that by design?

Reply:
Good question - it is by design.  That test case is a request for a first page of results, so `nextToken` isn't needed.  For requesting subsequent pages, the `nextToken` variable is added in `AppSyncPaginatedResultDeserializer.java` when creating the `requestForNextPage`.  

Prior to this refactor, it was included within the request document String because the `requestForNextPage` used the same request document `String`, and just set the `nextToken` in the `variables` json object.  This is possible now because `GraphQLRequest` now knows how to add inputs and regenerate a document String after the object is created.